pandas/core/arrays/integer.py

```diff
@@ -187,6 +187,9 @@ def coerce_to_array(values, dtype, mask=None, copy=False):
             raise TypeError("{} cannot be converted to an IntegerDtype".format(
                 values.dtype))
 
+    elif is_bool_dtype(values) and is_integer_dtype(dtype):
```

Comment:
actually, just check this if if ``dtype is not None``

Reply:
Like this?
```suggestion
    elif is_bool_dtype(values) and dtype is not None:
```

Reply:
yes, does that work?

Reply:
Yes, tests are okay.

Reply:
BTW, doesn't change the case `integer_array(array, dtype=None)`, though.

Reply:
can you show an explicit case when this fails: integer_array(array, dtype=None)

Reply:
`integer_array([True, False], dtype=None)`. You see, this will fail here on line 193 https://github.com/pandas-dev/pandas/blob/d0acd206a59279f9cea9615bfc8b9279273d495c/pandas/core/arrays/integer.py#L190-L195
because previous conditions won't trigger while target dtype is None. On the other hand we can't cast every boolean array to int because there're cases when we want this TypeError to trigger, as outlined in my previous [comment](https://github.com/pandas-dev/pandas/pull/25265#issuecomment-475614024)

Reply:
again can you modify the check so that a ``dtype=None`` case doesn't matter for boolean? write a test and see what needs modification while avoiding breaking other code

Reply:
See my latest commit, should work for this case, but kind of ugly.